draft-vkrasnov-h2-compression-dictionaries.md

```diff
@@ -103,10 +103,73 @@ document are to be interpreted as described in RFC 2119 {{RFC2119}}.
 
 ## Security Considerations
 
-The use of compression over an encrypted connection is known to leak
-potentialy sensitive information. We will collaborate with industry experts to
-identify any additional attack vectors introduced by this draft, and include
-a set of best practices to both servers and clients that would implement it.
+The use of compression over an encrypted connection could be used by malicious
+actors to potentially leak sensitive information. We will collaborate with
+industry experts to identify any additional attack vectors introduced by this
+draft, and include a set of best practices to both servers and clients that
+would implement it.
+
+### Attack scenarios and mitigations
+A single HTTP/2 connection is likely to be shared among multiple origins (over
+which it is authoritative) and among different navigation contexts to the same
+origin. When such sharing happens, and if compression contexts are shared
+between those instances, an attacker can use a BREACH-style attack in order to
+exfiltrate secrets from the context. Such secrets may include:
+
+* Cookies set using Javascript (and in-particular `httponly` cookies set from
+anonymous functions in external JS, which is not accessible to scripts
+otherwise)
+* CSRF tokens
+* CSP nonces
+* Application level secrets (e.g. financial information, stored credit cards
+numbers, codes, etc.)
+
+The mechanism for such data theft can happen if the attacker can:
+* Download multiple similar payloads to the target page modulo the actual
+secret, while trying out multiple permutations of the secret.
+* Observe the on-the-wire transfer size using Resource Timing's
```

Comment:
What about passive network monitoring?

Reply:
I don't think it's a threat here, but I could be wrong